code/game/objects/items/devices/healthanalyzerpro.dm

```diff
@@ -0,0 +1,425 @@
+/obj/item/device/healthanalyzerpro/
+	name = "ProHealth Analyzer"
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/misc_tools.dmi', "right_hand" = 'icons/mob/in-hand/right/misc_tools.dmi')
+	icon = 'icons/obj/device.dmi'
+	icon_state = "adv_health"
+	item_state = "healthanalyzer"
+	desc = "A hand-held body scanner able to precisely distinguish vital signs of the subject. This particular device is an experimental model outfitted with several modules that fulfill the roles of common scanning tools, memory function to record last made scan and a printer."
+	flags = FPRINT
+	siemens_coefficient = 1
+	slot_flags = SLOT_BELT
+	throwforce = 5
+	w_class = W_CLASS_MEDIUM
+	throw_speed = 4
+	starting_materials = list(MAT_IRON = 700, MAT_PLASTIC = 200, MAT_URANIUM = 50, MAT_SILVER = 50, MAT_GOLD = 50)
+	w_type = RECYK_ELECTRONIC
+	melt_temperature = MELTPOINT_PLASTIC
+	origin_tech = Tc_MAGNETS + "=4;" + Tc_BIOTECH + "=4"
+	attack_delay = 0
+	var/last_scantime = 0
+	var/last_reading = null
+	var/mode = "Health Scan"
+	var/list/modes = list("Health Scan", "Simplified Health Scan", "Advanced Health Scan", "Reagents Scan", "Immunity Scan", "Autopsy Scan")
+	var/obj/item/device/antibody_scanner/immune
+	var/last_print
+	parent_type = /obj/item/weapon/autopsy_scanner
+
+/obj/item/device/healthanalyzerpro/examine(mob/user)
+	..()
+	to_chat(user, "<span class='notice'>Current active mode: [mode].</span>")
+
+/obj/item/device/healthanalyzerpro/verb/toggle_mode()
+	set name = "Switch mode"
+	set category = "Object"
+
+	mode = input(usr, "Please select module.", "Pro Health Scanner") in modes
+	last_reading = null
+	last_scantime = 0
+
+/obj/item/device/healthanalyzerpro/verb/print(mob/user)
+	set name = "Print from memory"
+	set category = "Object"
+
+	if(mode == "Immunity Scan")
+		to_chat(user, "<span class='warning'>Due to memory constraints, immunity scan doesn't provide printing function!</span>")
+		return
+	if(!last_reading)
+		to_chat(user, "<span class='warning'>The memory is empty.</span>")
+		return
+	if(!user.dexterity_check()) //it's a complex thingy
+		to_chat(user, "<span class='warning'>You don't have the dexterity to do this!</span>")
+		return
+	if(last_print + 150 > world.time)
+		to_chat(user, "<span class='warning'>Printing energy spent, please wait a moment.</span>")
+		return
+
+	for(var/mob/O in viewers(usr))
+		O.show_message("<span class='warning'>\the [src] rattles and prints out a sheet of paper.</span>", 1)
+	sleep(10)
+	last_print = world.time
+	var/obj/item/weapon/paper/R = new(loc)
+	R.name = "paper - '[mode] results'"
+	R.info = last_reading
+	if(istype(user,/mob/living/carbon))
```

Comment:
iscarbon() would be preferred here, though it's probably not necessary to check at all